src/Http/Routing/perf/Matching/MatcherSingleEntryBenchmark.cs

```diff
@@ -68,12 +64,11 @@ public async Task Dfa()
         [Benchmark]
         public async Task LegacyTreeRouter()
         {
-            var feature = _feature;
-
             var httpContext = Requests[0];
+            var feature = new EndpointSelectorContext(httpContext);
 
-            // This is required to make the legacy router implementation work with global routing.
-            httpContext.Features.Set<IEndpointFeature>(feature);
+            //// This is required to make the legacy router implementation work with global routing.
+            //httpContext.Features.Set<IEndpointFeature>(feature);
```

Comment:
I think this can just be removed right? Now this feature is always present.

Reply:
Yea I wasn’t sure at first

Reply:
It's ok. we all feel that way sometimes.